org.alloytools.alloy.application/src/main/java/edu/mit/csail/sdg/alloy4whole/PreferencesDialog.java

```diff
@@ -312,13 +331,42 @@ private static boolean isSat(String output) {
         return output.substring(i).startsWith("s SATISFIABLE");
     }
 
+    // [HASLab]
+    private static boolean staticLibrary(String name) {
+        // check if in java library path
+        final String[] dirs = System.getProperty("java.library.path").split(System.getProperty("path.separator"));
+        for (int i = dirs.length - 1; i >= 0; i--) {
+            final File file = new File(dirs[i] + File.separator + name);
+            if (file.canExecute()) {
+                if ("yes".equals(System.getProperty("debug")))
```

Comment:
> `"yes".equals(System.getProperty("debug"))`

I know it doesn't matter that much, but it looks silly to recompute this in every single loop iteration; compute it once once (in a static final variable, or at the beginning of the method) and use the result in all these `if`s